### PR TITLE
Documentation on how to migrate to new indexer

### PR DESCRIPTION
**Pull request by rtso**
Add a one pager with instructions on how to migrate from indexer fullnode to new indexer stack

## Testing
Followed this guide: https://www.notion.so/aptoslabs/Indexer-Docs-Writing-docs-3b8b6ed8d28b45248ddbc0263b92f72b

### CODE REVIEW

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: "Migrate to Transaction Stream Service"
+---
+
+This guide contains information on how to migrate to using the Transaction Stream Service if you are currently running a legacy indexer.
+
+The main benefit of using the Transaction Stream Service is that you no longer need to be running an indexer fullnode. More documentation on the Transaction Stream Service is provided [here](https://aptos.dev/indexer/txn-stream/).
```
Comment (bowenyang007) [inline developer-docs-site/docs/indexer/legacy/migration.md:7]: you no longer need to run an indexer fullnode

**Review by bowenyang007 — APPROVED** [review]

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+---
+title: "Migrate to Transaction Stream Service"
+---
+
+This guide contains information on how to migrate to using the Transaction Stream Service if you are currently running a legacy indexer.
+
+The old set up requires running an archival fullnode with additional threads to process the transactions which is difficult and expensive to maintain. Adding more custom logic either requires a bulkier machine, or running several fullnodes that scale linearly.
+
+This new way of indexing requires only a single (or 2 for redundancy) fullnodes. Custom logic live in separate light servers that scale separately from the fullnodes so you don’t need to run multiple fullnodes. You can also directly stream from Labs provided endpoints which means that you no longer need to run fullnodes.
```
Comment (banool) [inline developer-docs-site/docs/indexer/legacy/migration.md:9]: I wouldn't mention that users have to run fullnodes here, just mention that they have to use the Transaction Stream Service and briefly mention that they can either use our hosted, ratelimited instance or run their own. No need to go too deep since you dive into that choice later on.

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  postgres_connection_string: postgresql://postgres:@localhost:5432/indexer
+  indexer_grpc_data_service_address: 127.0.0.1:50051
+  indexer_grpc_http2_ping_interval_in_secs: 60
+  indexer_grpc_http2_ping_timeout_in_secs: 10
+  auth_token: AUTH_TOKEN
+```
+
+To connect the processor to the Transaction Stream Service, you need to set the URL for `indexer_grpc_data_service_address`. Choose one of the following options.
+
+### Option a: Connect to Labs-provided Transaction Stream Service
```
Comment (banool) [inline developer-docs-site/docs/indexer/legacy/migration.md:45]: Nit:
```suggestion
### Option A: Connect to Labs-provided Transaction Stream Service
```

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+By using the Labs-provided Transaction Stream Service, you no longer have to run an archival node. Instructions to connect to Labs-provided Transaction Stream can be found [here](https://aptos.dev/indexer/txn-stream/labs-hosted).
```
Comment (banool) [inline developer-docs-site/docs/indexer/legacy/migration.md:47]: Nit, elsewhere we call it `Labs-hosted`
```suggestion
By using the Labs-hosted Transaction Stream Service, you no longer have to run an archival node. Instructions to connect to Labs-provided Transaction Stream can be found [here](https://aptos.dev/indexer/txn-stream/labs-hosted).
```

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+### Option b: Run a Self-Hosted Transaction Stream Service
+
+If you choose to, you can run a self-hosted instance of the Transaction Stream Service and connect your processors to it. Instructions to run a Self-Hosted Transaction Stream can be found [here](https://aptos.dev/indexer/txn-stream/self-hosted).
+
+## 3. (Optional) Migrate custom processors to Transaction Stream Service
+
+If you have custom processors written with the old indexer, we highly recommend starting from scratch and using a new database than what your legacy indexer is already writing to. Using a new database ensures that all your custom database migrations will be applied during this migration.
```
Comment (banool) [inline developer-docs-site/docs/indexer/legacy/migration.md:55]: Nit:
```suggestion
If you have custom processors written with the old indexer, we highly recommend starting from scratch with a new database. Using a new database ensures that all your custom database migrations will be applied during this migration.
```

File: developer-docs-site/docs/indexer/legacy/migration.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+```
+Processor::CoinProcessor => {
+    Arc::new(CoinTransactionProcessor::new(self.db_pool.clone()))
+},
+```
+
+## 4. Backfill Postgres database with Diesel
+
+Even though the new processors have the same Postgres schemas as the old ones, we recommend to do a complete backfill (ideally writing to a new DB altogether) because some fields are a bit different as a result of the protobuf conversion.
```
Comment (banool) [inline developer-docs-site/docs/indexer/legacy/migration.md:91]:
```suggestion
Even though the new processors have the same Postgres schemas as the old ones, we recommend you do a complete backfill (ideally writing to a new DB altogether) because some fields are a bit different as a result of the protobuf conversion.
```

**Review by banool — APPROVED** [review]
Looks good!

The guide is maybe a little hard to follow but since this doc is for people already familiar with the legacy indexer, I think this is fine.

**Comment by rtso** [comment]
@banool this doc assumes they've read and understand all the other docs under [Legacy Indexer](https://aptos.dev/indexer/legacy/).
